#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.29035551341129817 0.7237615545957902 0.6259895270056411 3.8272485192068255
  position -0.20923967213231678 2.19393039323369 -0.04574681858872337
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
WoodenBox {
  translation -0.467082 0.05 0.323801
  rotation 0 1 0 -0.261795307179586
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.21 0.05 0.45
  name "wooden box(8)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.337574 0.05 0.396272
  rotation 0 1 0 0.785398
  name "wooden box(9)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.44 0.05 -0.3
  name "wooden box(10)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.28 0.05 -0.44
  name "wooden box(2)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.35284 0.05 -0.417278
  rotation 0 -1 0 0.785396
  name "wooden box(3)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.411213 0.05 -0.387071
  rotation 0 1 0 0.785398
  name "wooden box(4)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.02 0.05 0.22
  name "wooden box(5)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.43 0.05 0.23732
  rotation 0 1 0 -0.523595307179586
  name "wooden box(7)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.275858 0.05 0.424142
  rotation 0 1 0 -0.785395307179586
  name "wooden box(1)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.37 0.05 0.33
  name "wooden box(6)"
  size 0.1 0.1 0.1
}
DEF WheelBot Robot {
  translation 0.22870946315941754 -0.0003164401169611364 -0.039999926795210944
  rotation 0.02901619155320393 0.9991577078406753 0.02901612466053425 -1.5716373070313356
  children [
    DEF WHEELBOTBODYSHAPE Transform {
      translation 0 0.04 0
      rotation 1 0 0 0
      children [
        Shape {
          appearance DEF BODYCOLOUR PBRAppearance {
            baseColor 0.4 0 0.3
            metalness 0
            IBLStrength 0.2
          }
          geometry DEF WHEELBOTBODY Cylinder {
            height 0.075
            radius 0.04
          }
        }
      ]
    }
    DEF COMPASS Group {
      children [
        Transform {
          translation 0 0.0775 0.011
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance DEF WHEELCOLOR PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Cone {
                bottomRadius 0.016
                height 0.05
                subdivision 14
              }
            }
          ]
        }
        Transform {
          translation 0 0.0775 -0.02
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                metalness 0
              }
              geometry Capsule {
                height 0.001
                radius 0.017
              }
            }
          ]
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -5.247999915362059
        axis -1 0 0
        anchor 0 0.02 0
      }
      device [
        RotationalMotor {
          name "leftmotor"
        }
      ]
      endPoint DEF WHEELBOTWHEEL Solid {
        translation 6.299322153416525e-09 0.009796419421954217 0.01719238582165944
        rotation 0.9999999999999508 -1.6306257928128134e-07 -2.6809558582869865e-07 5.2479998956575376
        children [
          DEF LEFTWHEELSHAPE Transform {
            translation 0.0425 0.02 0
            rotation 0 0 1 1.5707996938995747
            children [
              DEF WHEELSHAPE Shape {
                appearance DEF WHEELCOLOR PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry DEF WHEELBOTWHEELSHAPE Cylinder {
                  height 0.008
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "leftwheel"
        boundingObject USE LEFTWHEELSHAPE
        physics DEF WHEELPHYSICS Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 5.247999916616296
        anchor 0 0.02 0
      }
      device [
        RotationalMotor {
          name "rightmotor"
          maxTorque 30
        }
      ]
      endPoint Solid {
        translation -6.30162775095803e-09 0.009796419104501818 0.017192386323534037
        rotation 0.9999999999999493 1.6529133179985475e-07 2.717598388982212e-07 5.247999896641229
        children [
          DEF RIGHTWHEELSHAPE Transform {
            translation -0.0425 0.02 0
            rotation 0 0 1 1.5707996938995747
            children [
              USE WHEELSHAPE
            ]
          }
        ]
        name "rightwheel"
        boundingObject USE RIGHTWHEELSHAPE
        physics USE WHEELPHYSICS
      }
    }
    RangeFinder {
      translation 0 0.035 0.039
      rotation 0 1 0 3.141592653589793
      children [
        DEF RFSHAPE Transform {
          translation 0 0 -0.001
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance USE WHEELCOLOR
              geometry Cylinder {
                height 0.001
                radius 0.002
              }
            }
          ]
        }
      ]
      name "range finder"
      boundingObject USE RFSHAPE
      physics Physics {
      }
      fieldOfView 1.5
      width 300
      height 200
    }
    Radar {
      translation 0 0.04 1.73472e-18
      rotation 0 1 0 3.14159
      minRange 0.01
      maxRange 1.5
      horizontalFieldOfView 1
      verticalFieldOfView 0.3
      occlusion TRUE
    }
  ]
  name "wheelbot"
  boundingObject USE WHEELBOTBODYSHAPE
  physics Physics {
    density -1
    mass 0.5
    centerOfMass [
      0 0.04 0
    ]
    inertiaMatrix [
      0.0009306968236259764 0.00030159289474462024 0.0009306968236259764
      0 0 0
    ]
  }
  controller "RadarController"
}
DEF WheelBot Robot {
  translation -0.250308 0.0333414 -0.32
  rotation 1.1646699987738426e-06 4.5871899951706346e-05 0.9999999989472061 -0.0580653071795858
  children [
    DEF WHEELBOTBODYSHAPE Transform {
      translation 0 0.04 0
      rotation 1 0 0 0
      children [
        Shape {
          appearance DEF BODYCOLOUR PBRAppearance {
            baseColor 0.4 0 0.3
            metalness 0
            IBLStrength 0.2
          }
          geometry DEF WHEELBOTBODY Cylinder {
            height 0.075
            radius 0.04
          }
        }
      ]
    }
    DEF COMPASS Group {
      children [
        Transform {
          translation 0 0.0775 0.011
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance DEF WHEELCOLOR PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Cone {
                bottomRadius 0.016
                height 0.05
                subdivision 14
              }
            }
          ]
        }
        Transform {
          translation 0 0.0775 -0.02
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
                metalness 0
              }
              geometry Capsule {
                height 0.001
                radius 0.017
              }
            }
          ]
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -5.247999915362059
        axis -1 0 0
        anchor 0 0.02 0
      }
      device [
        RotationalMotor {
          name "leftmotor"
        }
      ]
      endPoint DEF WHEELBOTWHEEL Solid {
        translation 6.299322153416535e-09 0.009796419421954215 0.01719238582165945
        rotation 0.9999999999999503 -1.6380886494829276e-07 -2.6932257422772205e-07 5.2479998955673555
        children [
          DEF LEFTWHEELSHAPE Transform {
            translation 0.0425 0.02 0
            rotation 0 0 1 1.5707996938995747
            children [
              DEF WHEELSHAPE Shape {
                appearance DEF WHEELCOLOR PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry DEF WHEELBOTWHEELSHAPE Cylinder {
                  height 0.008
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "leftwheel"
        boundingObject USE LEFTWHEELSHAPE
        physics DEF WHEELPHYSICS Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 5.247999916616296
        anchor 0 0.02 0
      }
      device [
        RotationalMotor {
          name "rightmotor"
          maxTorque 30
        }
      ]
      endPoint Solid {
        translation -6.301627750958037e-09 0.009796419104501797 0.017192386323534082
        rotation 0.9999999999999476 1.6821706916807396e-07 2.7657012088450516e-07 5.247999896287659
        children [
          DEF RIGHTWHEELSHAPE Transform {
            translation -0.0425 0.02 0
            rotation 0 0 1 1.5707996938995747
            children [
              USE WHEELSHAPE
            ]
          }
        ]
        name "rightwheel"
        boundingObject USE RIGHTWHEELSHAPE
        physics USE WHEELPHYSICS
      }
    }
    RangeFinder {
      translation 0 0.035 0.039
      rotation 0 1 0 3.141592653589793
      children [
        DEF RFSHAPE Transform {
          translation 0 0 -0.001
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance USE WHEELCOLOR
              geometry Cylinder {
                height 0.001
                radius 0.002
              }
            }
          ]
        }
      ]
      name "range finder"
      boundingObject USE RFSHAPE
      physics Physics {
      }
      fieldOfView 1.5
      width 300
      height 200
    }
    Radar {
      translation 0 0.04 1.73472e-18
      rotation 0 1 0 3.14159
      minRange 0.01
      maxRange 1.5
      horizontalFieldOfView 1
      verticalFieldOfView 0.3
      occlusion TRUE
    }
  ]
  name "wheelbot(1)"
  boundingObject USE WHEELBOTBODYSHAPE
  physics Physics {
    density -1
    mass 0.5
    centerOfMass [
      0 0.04 0
    ]
    inertiaMatrix [
      0.0009306968236259764 0.00030159289474462024 0.0009306968236259764
      0 0 0
    ]
  }
  radarCrossSection 1
  controller "RadarController"
}
Robot {
  name "Supervisor"
  controller "SupervisorController"
  supervisor TRUE
}
